src/elin_bjoernson_ex/ex05/bounded_sim.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+from walker_sim import Walker, Simulation
+import random
+
+__author__ = 'Elin Wølner Bjørnson'
+__email__ = 'elinbj@nmbu.no'
+
+
+class BoundedWalker(Walker):
+    def __init__(self, start, home, left_limit, right_limit):
+        """
+        Initialise the walker
+
+        Arguments
+        ---------
+        start : int
+            The walker's initial position
+        home : int
+            The walk ends when the walker reaches home
+        left_limit : int
+            The left boundary of walker movement
+        right_limit : int
+            The right boundary  of walker movement
+        """
+        self.left_limit = left_limit
+        self.right_limit = right_limit
+
+        super().__init__(start, home)
+
+    def move(self):
+        if self.position >= self.right_limit:
+            self.position -= 1
+        elif self.position <= self.left_limit:
+            self.position += 1
+        else:
+            left_or_right = [-1, 1]
+            self.position += random.choice(left_or_right)
+            self.moves += 1
```

Comment:
Let's say at the start of this function:
self.right_limit = 10
self.position = 10
self.move = 3
After `move()` is finished:
self.right_limit = 10
self.position = 9
self.move = 3
When `move()` again, and the walker move left:
self.right_limit = 10
self.position = 8
self.move = 4

So the walker moves 2 steps in one move, which is not really what the walker should do.